cli/src/pcluster/models/cluster_resources.py

```diff
@@ -95,6 +96,91 @@ def _is_failed_wait(event):
+                "OnNodeConfiguredExecutionFailure",
+                "Failed to execute OnNodeConfigured script.",
+                "Failed to execute OnNodeConfigured script",
+            ),
+            ClusterCreationFailure(
+                "OnNodeConfiguredDownloadFailure",
+                "Failed to download OnNodeConfigured script.",
+                "Failed to download OnNodeConfigured script",
+            ),  # s3 and wget
+            ClusterCreationFailure(
+                "OnNodeConfiguredFailure",
+                "Failed to run OnNodeConfigured script.",
+                "Failed to run OnNodeConfigured script",
+            ),
+            ClusterCreationFailure(
+                "OnNodeStartExecutionFailure",
+                "Failed to execute OnNodeStart script.",
+                "Failed to execute OnNodeStart script",
+            ),
+            ClusterCreationFailure(
+                "OnNodeStartDownloadFailure",
+                "Failed to download OnNodeStart script.",
+                "Failed to download OnNodeStart script",
+            ),  # s3 and wget
+            ClusterCreationFailure(
+                "OnNodeStartFailure", "Failed to run OnNodeStart script.", "Failed to run OnNodeStart script"
+            ),
+            ClusterCreationFailure("EbsMountFailure", "Failed to mount EBS volume.", "Failed to mount EBS volume"),
+            ClusterCreationFailure("EfsMountFailure", "Failed to mount EFS.", "Failed to mount EFS"),
+            ClusterCreationFailure("FsxMountFailure", "Failed to mount FSX.", "Failed to mount FSX"),
+            ClusterCreationFailure("RaidMountFailure", "Failed to mount RAID array.", "Failed to mount RAID array"),
+            ClusterCreationFailure(
+                "HeadNodeBootstrapFailure", "Failed to set up the head node.", "configured scheduler plugin"
+            ),
+            ClusterCreationFailure(
+                "HeadNodeBootstrapFailure", "Failed to set up the head node.", "Failed to run chef recipe"
+            ),
+            ClusterCreationFailure(
+                "HeadNodeBootstrapFailure", "Failed to bootstrap the head node.", "Failed to bootstrap the head node"
+            ),
+            ClusterCreationFailure(
+                "ResourceCreationFailure",
+                "Failed to create resources for head node bootstrap.",
+                "Resource creation cancelled",
+            ),
+            ClusterCreationFailure(
+                "HeadNodeBootstrapFailure", "Cluster creation timed out.", "WaitCondition timed out"
+            ),
+        ]
+
+        general_failure = failure = ClusterCreationFailure(
+            "ClusterCreationFailure", "Failed to create the cluster.", ""
+        )
+        cfn_failure_reason = self._get_failure_reason()
+        if cfn_failure_reason:
+            failure = next(
+                (f for f in cluster_creation_failures if f.cfn_failure_reason in cfn_failure_reason), general_failure
```

Comment:
question, this works well only if all `cfn_failure_reason` are unique, should we enforce this?

Reply:
yes, currently the cfn failure reasons are start_with some unique strings. With different `cfn_failure_reasons`, we map it for API failure reason and code